### PR TITLE
feat(core/menu): implement a11y features for menu and burger-menu

### PR DESCRIPTION
**Pull request by jg0815**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`yarn build`) was run locally and any changes were pushed
- [X] Unit tests (`yarn test`) were run locally and passed
- [ ] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [X] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [X] Bug fix
- [X] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [X] Documentation content changes
- [ ] Other (please describe):

## What is the new behavior?

- Implemented keyboard accessibility of the &lt;ix-burger-menu&gt; component by inserting a &lt;button&gt; element.
- Implemented new property 'ixAriaLabel' for the &lt;ix-burger-menu&gt; component, which is required as an accessibility label for the respective icon button.
- Set the ARIA-attributes 'aria-label' and  'aria-pressed' for the &lt;ix-burger-menu&gt; component.
- Adjusted focus frame settings of the &lt;ix-burger-menu&gt; component via CSS.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: packages/core/src/components/menu/burger-menu.scss [hunk 1]
```diff
@@ -9,37 +9,76 @@
...
+    border-width: $focus-bdr-width;
+    border-style: solid;
+    border-color: transparent;
+    border-radius: var(--theme-btn--border-radius);
+    outline: none;
+    cursor: pointer;
+  }
+}
+
+:host(:focus) {
```
Comment (nuke-ellington) [inline packages/core/src/components/menu/burger-menu.scss:61]: should be :focus-visible (this also causes the VRTs to fail)

File: packages/core/src/components/menu/burger-menu.scss [hunk 2]
```diff
@@ -9,37 +9,76 @@
...
+  }
 }
 
 :host(:active) {
-  background-color: var(--theme-menu-btn--background--active);
-  border-radius: var(--theme-btn--border-radius);
+  .burger-menu-button {
+    background-color: var(--theme-menu-btn--background--active);
+    border-width: $focus-bdr-width;
+    border-style: solid;
```
Comment (nuke-ellington) [inline packages/core/src/components/menu/burger-menu.scss:76]: There should not be a border on the active button

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/components/menu/burger-menu.scss [hunk 3]
```diff
@@ -9,37 +9,76 @@
...
+:host(:focus) {
+  .burger-menu-button {
+    border-width: $focus-bdr-width;
+    border-style: solid;
+    border-color: var(--theme-color-focus-bdr);
+    border-radius: var(--theme-btn--border-radius);
+    outline: none;
+    cursor: pointer;
+  }
 }
```
Comment (nuke-ellington) [inline packages/core/src/components/menu/burger-menu.scss:70]:
```suggestion
.burger-menu-button:focus-visible {
  border-width: $focus-bdr-width;
  border-style: solid;
  border-color: var(--theme-color-focus-bdr);
  border-radius: var(--theme-btn--border-radius);
  outline: none;
  cursor: pointer;
}
```

**Review by nuke-ellington — COMMENTED** [review]
Something like this should do the trick

File: packages/core/src/components/menu/burger-menu.scss (hunk 3 above)
Comment (jg0815) [reply packages/core/src/components/menu/burger-menu.scss:70]: Regrettably, not completely: After applying this change, the focus frame is **not** displayed during the mouse click, but it **is** displayed afterwards. I think that this is not the desired behavior?

Comment (nuke-ellington) [reply packages/core/src/components/menu/burger-menu.scss]: If I run it locally there is no focus frame when interacting via mouse. Only when navigating by keyboard :)

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**